Bring win_lgpo changes in from master

twangboy opened the pull request:
### What does this PR do?
Brings in changes from PR to master: https://github.com/saltstack/salt/pull/54896

### Tests written?
Yes

### Commits signed with GPG?
Yes

twangboy committed:
Bring win_lgpo changes in from master

Akm0d commented:
Let's split this into 2 or 3 PRs; it's too huge.  `.ci` changes in one, `requirements` changes in another, and `salt`/`tests` changes in another

twangboy committed:
Add ability to handle Not Defined for REG_MULTI_SZ

twangboy committed:
Fix read_value function in win_reg salt util

An empty REG_MULTI_SZ value was returning None instead of an empty list.

twangboy committed:
Fix spelling error

twangboy committed:
Handle xml with invalid xmlns entries

- WindowsDefender.adml

twangboy committed:
Fix variable names

twangboy committed:
Fix some comments

twangboy committed:
Fix issue with text fields

twangboy committed:
Speed optimizations... possibly

Set the key to lowercase in the admx/adml files and cache those
Change the REGKEY_XPATH back to no longer check for case
Needs thorough testing

xeacott commented:
This is gunna be amazing.

twangboy committed:
Add chardet dependency

twangboy committed:
More LGPO Optimization

Adds a new function `get_policy` that allows you to get the current
setting for a single policy on the machine.
Adds some helper functions to get machine settings for policies defined
in `_policy_info` (`_get_policy_info_setting`) and policies defined in
the policy templates (`_get_policy_adm_setting`)
Changes the `win_lgpo` state function to use `get_policy` instead of
listing all policies and their settings on the machine. This is a huge
speed increase.

twangboy committed:
Fix some issues with the get_policy function

twangboy committed:
Remove chardet dependency
